demos/mantle_convection/dynamic_topography/dynamic_topography.py

```diff
@@ -0,0 +1,141 @@
+# Tutorial: Computing Normal Stresses and Dynamic Topography in G-ADOPT
+# ====================================================================
+# Overview
+# --------
+# One of the most commonly studied geodynamic observables is **dynamic topography** —
+# the surface deformation caused by internal stresses within the Earth's mantle.
+# This tutorial shows how to compute **normal stresses acting on a boundary**
+# using **G-ADOPT**.
+# Specifically, we will compute the radial stress $\sigma_{rr}$ on a boundary and
+# use it to estimate dynamic topography.
+
+# Theory Refresher
+# ----------------
+# Dynamic topography arises from the internal stress field deforming a surface.
+# Consider the top boundary of an annular domain located at radius $r_{max}$.
+# Under equilibrium, the vertical (radial) stress acting on this boundary is $\sigma_{rr}(r_max)$.
+# Due to internal forces, the surface deforms by an amount $\delta h$, leading to:
+#
+# $$\sigma_{rr}(r_max) = \sigma_{rr}(r_max + \delta h) - \rho g\, \delta h$$
+#
+# Solving for $\sigma h$:
+#
+# $$\delta h = \sigma_{rr} / (\rho g)$$
+#
+# Where:
+# - $\sigma_{rr}$ is the normal stress at the boundary,
+# - $\rho$ is the density difference between the mantle and the overlying medium (air or water),
+# - $g \approx 9.8 m/s^2$ is gravitational acceleration.
+
+# Implementation in G-ADOPT
+# -------------------------
+
+# Step 1: Load Mesh and Temperature Field
```

Comment:
The different steps here, for consistency with other tutorials, put them in separate blocks? Using + and -?